src/background/services/paymentSession.ts

```diff
@@ -300,4 +307,61 @@ export class PaymentSession {
     this.amount = amount.toString()
     this.intervalInMs = Number((amount * BigInt(HOUR_MS)) / BigInt(this.rate))
   }
+
+  private async getOverpayingSession(): Promise<{
+    session: OverpayingSession | null
+    sessionIndex: number
+  }> {
+    const { overpayingSessions } = await this.storage.get([
+      'overpayingSessions'
+    ])
+
+    const hashUrl = await getHashUrl(this.url || '')
```

Comment:
When `this.url` could be a falsy btw?

Reply:
I'm using the url from the sender of browser.runtime, which is optional